synapse/rest/admin/rooms.py

```diff
@@ -253,6 +253,43 @@ async def on_GET(self, request):
         return 200, response
 
 
+class RoomRestServlet(RestServlet):
+    """Get room details.
+    This needs user to have administrator access in Synapse.
+    TODO: Return power level for each user
+    TODO: Add on_POST to allow room creation without joining the room
+
+    GET /_synapse/admin/v1/rooms/<room_id>
+
+    returns:
+        200 OK with users if success otherwise an error.
+    """
+
+    PATTERNS = admin_patterns("/rooms/(?P<room_id>[^/]+)$")
+
+    def __init__(self, hs):
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, room_id):
+        requester = await self.auth.get_user_by_req(request)
+        await assert_user_is_admin(self.auth, requester.user)
+
+        ret = await self.store.get_room_stats_state(room_id)
+        if not ret:
+            raise NotFoundError("Room not found")
+
+        stats_curr = await self.store.get_room_stats_current(room_id)
+        if stats_curr:
+            ret.update(stats_curr)
+
+        members = await self.store.get_users_in_room(room_id)
+        ret["members"] = members if members else []
```

Comment:
I believe that `get_users_in_room` will always return a list, so I think we can just set `ret["members"] = members` here.